microceph/ceph/config.go

```diff
@@ -162,43 +163,104 @@ func ListConfigs() (types.Configs, error) {
 	return configs, nil
 }
 
-// updates the ceph config file.
-func UpdateConfig(s interfaces.StateInterface) error {
-	confPath := filepath.Join(os.Getenv("SNAP_DATA"), "conf")
-	runPath := filepath.Join(os.Getenv("SNAP_DATA"), "run")
+// backwardCompatPubnet ensures that the public_network is set in the database
+// this is a backward-compat shim to accomodate older versions of microceph
+// which will ensure that the public_network is set in the database
+func backwardCompatPubnet(s interfaces.StateInterface) error {
+	config, err := getConfigDb(s)
+	if err != nil {
+		return fmt.Errorf("failed to get config from db: %w", err)
+	}
 
-	// Get the configuration and servers.
+	// do we have a public_network configured?
+	pubNet := config["public_network"]
```

Comment:
NIT: Should we not check if the key exists ?
```
val, ok := myMap["foo"]
if !ok {
    // handle absence of public_network.
}
```

Reply:
The reasoning is if we don't have a pub net, config["public_network"] will evaluate to the empty string, which will fail the ParseCIDR() coming below

Reply:
ah okay, I was assuming this to be the case. Can we just add a short comment which describes this in place of the `do we have a public_network configured?` comment ? I think that could be helpful.